### ISSUE TITLE
Combine Server-only rendering with internal frontend components.

### ISSUE THREAD

**Issue by davidmeirlevy** [nuxt/nuxt#6609]
### What problem does this feature solve?

1. Reduce to minimum the `window.__NUXT__` object.
At the fully server-side rendered website, that SEO is a key issue - this object is an obstacle.

2. Don't really need frontend components in most of the app.
Most pages and components can render by server, and not be re-managed by frontend, and don't need the hydration there.

3. There are still cases we do need frontend components.
Even inside an app like I mentioned, there's still a component that I want to manage from frontend (hydration or not, is not the key problem in those cases, like: charts, graphs, live social interactions, etc.)

### What does the proposed changes look like?

A mixed mode for nuxt, that I can "mark" a component as "only for server" or "important for frontend" (choose what you like), or something similar to "v-once" directive of vue, that renders only once, but to render it on server and ignore the hydration for those components.
so every data that is used for those "once" components that will be rendered only on the server side, will not appear inside the `__NUXT__` object.

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9938">#c9938</a>)</em></sub></div>

**Comment by aldarund** [comment]
Take a look at https://github.com/maoberlehner/vue-lazy-hydration

**Comment by davidmeirlevy** [comment]
> Take a look at https://github.com/maoberlehner/vue-lazy-hydration

@aldarund 
Seems magical and cool, i'll check it as soon as possible!
does it handle the "nuxt" global object to remove data that used for ssr-only components?

thanks.

**Comment by testn** [comment]
any update on this?

**Comment by danielroe** [comment]
We are approaching the Nuxt 2 EOL date (June 30, 2024) - see [this article](https://nuxt.com/blog/nuxt2-eol) for more information. I'm closing this issue as it's marked as a _Nuxt 2 related enhancement_ and it's not a critical issue.

That doesn't mean it might not be relevant for Nuxt 3. If it is, please feel free to open a new issue (or just comment, and I can reopen it). 🙏

Thank you for your understanding and for your contribution to Nuxt! 🎉
